test/functional/test_framework/test_node.py

```diff
@@ -267,6 +283,29 @@ def assert_debug_log(self, expected_msgs):
                 if re.search(re.escape(expected_msg), log, flags=re.MULTILINE) is None:
                     self._raise_assertion_error('Expected message "{}" does not partially match log:\n\n{}\n\n'.format(expected_msg, print_log))
 
+    @contextlib.contextmanager
+    def assert_memory_usage_stable(self, perc_increase_allowed=0.03):
+        """Context manager that allows the user to assert that a node's memory usage (RSS)
+        hasn't increased beyond some threshold percentage.
+        """
+        before_memory_usage = self.get_mem_rss()
+
+        yield
+
+        after_memory_usage = self.get_mem_rss()
+
+        if not (before_memory_usage and after_memory_usage):
+            self.log.warning("Unable to detect memory usage (RSS) - skipping memory check.")
+            return
+
+        perc_increase_memory_usage = 1 - (float(before_memory_usage) / after_memory_usage)
+
+        if perc_increase_memory_usage > perc_increase_allowed:
+            self._raise_assertion_error(
+                "Memory usage increased over threshold of {}% from {} to {} ({}%)".format(
+                    perc_increase_allowed, before_memory_usage, after_memory_usage,
```

Comment:
Since you are displaying as a % instead of decimal, I think you just need to multiply by 100
```
.format(perc_increase_allowed*100, 
        before_memory_usage, 
        after_memory_usage, 
        perc_increase_memory_usage*100)
```

Now it'll show something like
`Memory usage increased over threshold of 3.0%....`